sklearn/preprocessing/target.py

```diff
@@ -0,0 +1,223 @@
+        the transformer will be an identity transformer. Note that the
+        transformer will be cloned during fitting.
+
+    func : function, optional
+
+        Function to apply to ``y`` before passing to ``fit``. Cannot be set at
+        the same time as ``transformer``. The function needs to return a
+        2-dimensional array. If ``func`` is ``None``, the function used will be
+        the identity function.
+
+    inverse_func : function, optional
+        Function to apply to the prediction of the regressor. Cannot be set at
+        the same time as ``transformer`` as well. The function needs to return
+        a 2-dimensional array. The inverse function is used to return
+        predictions to the same space of the original training labels.
+
+    check_inverse : bool, default=True
+        Whether to check that ``transform`` followed by ``inverse_transform``
+        or ``func`` followed by ``inverse_func`` leads to the original targets.
+
+    Attributes
+    ----------
+    regressor_ : object
+        Fitted regressor.
+
+    transformer_ : object
+        Transformer used in ``fit`` and ``predict``.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from sklearn.linear_model import LinearRegression
+    >>> from sklearn.preprocessing import TransformedTargetRegressor
+    >>> tt = TransformedTargetRegressor(regressor=LinearRegression(),
+    ...                               func=np.log, inverse_func=np.exp)
+    >>> X = np.arange(4).reshape(-1, 1)
+    >>> y = np.exp(2 * X).ravel()
+    >>> tt.fit(X, y) # doctest: +ELLIPSIS
+    TransformedTargetRegressor(...)
+    >>> tt.score(X, y)
+    1.0
+    >>> tt.regressor_.coef_
+    array([ 2.])
+
+    Notes
+    -----
+    Internally, the target ``y`` is always converted into a 2-dimensional array
+    to be used by scikit-learn transformers. At the time of prediction, the
+    output will be reshaped to a have the same number of dimension as ``y``.
+
+    See :ref:`examples/preprocessing/plot_transform_target.py
+    <sphx_glr_auto_examples_preprocessing_plot_transform_target.py> `.
+
+    """
+    def __init__(self, regressor=None, transformer=None,
+                 func=None, inverse_func=None, check_inverse=True):
+        self.regressor = regressor
+        self.transformer = transformer
+        self.func = func
+        self.inverse_func = inverse_func
+        self.check_inverse = check_inverse
+
+    def _fit_transformer(self, y):
+        if (self.transformer is not None and
+                (self.func is not None or self.inverse_func is not None)):
+            raise ValueError("'transformer' and functions 'func'/"
+                             "'inverse_func' cannot both be set.")
+        elif self.transformer is not None:
+            self.transformer_ = clone(self.transformer)
+        else:
+            if self.func is not None and self.inverse_func is None:
+                raise ValueError("When 'func' is not None, 'inverse_func'"
```

Comment:
not covered ;)

Reply:
Yep, I forgot to call `fit` in the test. Should be covered now.

Reply:
I'd prefer "When func is provided, inverse_func must also be provided": users don't set these things to None.